schema/query.go

```diff
@@ -354,25 +358,33 @@ func (e NotExist) Match(payload map[string]interface{}) bool {
 // Match implements Expression interface
 func (e GreaterThan) Match(payload map[string]interface{}) bool {
 	n, ok := isNumber(getField(payload, e.Field))
-	return ok && (n > e.Value)
+	if ok {
+		return ok && (n > e.Value.(float64))
+	}
+	t, ok := isTime(getField(payload, e.Field))
+	if ok {
```

Comment:
Thanks for the tips. Still getting used to the idioms in Go. In this block there are actually 2 type conversion checks (for both the field & filter values), so I'd like to get your feedback on the options to check each.  This stays more concise and inline with the original code:

```go
if t, ok := isTIme(getField(payload, e.Field)); ok {
    t2, ok2 := isTime(e.Value)
    return ok2 && t.After(t2)
}
```

This other seems more clear:

```go
if t, ok := isTIme(getField(payload, e.Field)); ok {
    if t2, ok2 := isTime(e.Value); ok2 {
        return t.After(t2)
    }
}
```

I'm leaning towards the 2nd option. Either is good by me.